include/ddc/kernels/splines/matrix_sparse.hpp

```diff
@@ -98,13 +98,22 @@ template <class ExecSpace>
 class Matrix_Sparse : public Matrix
 {
     using matrix_sparse_type = gko::matrix::Csr<double, int>;
+#ifdef KOKKOS_ENABLE_OPENMP
+    using solver_type = std::conditional_t<
+            std::is_same_v<ExecSpace, Kokkos::OpenMP>,
+            gko::solver::Gmres<double>,
+            gko::solver::Bicgstab<double>>;
+#else
+    using solver_type = gko::solver::Bicgstab<double>;
+#endif
+
 
 private:
     std::unique_ptr<gko::matrix::Dense<double>> m_matrix_dense;
 
     std::shared_ptr<matrix_sparse_type> m_matrix_sparse;
 
-    std::shared_ptr<gko::solver::Gmres<double>> m_solver;
+    std::shared_ptr<solver_type> m_solver;
```

Comment:
```suggestion
    std::shared_ptr<gko::LinOp> m_solver;
```

Will be even simpler to change the solver

Reply:
This does not compile:

![image](https://github.com/CExA-project/ddc/assets/22033714/3b3dbdd3-3271-4a63-940b-2ceb49367e6d)

Reply:
```
        std::shared_ptr const solver = solver_factory->generate(m_matrix_sparse);
        m_solver = solver;
        m_solver_tr = solver->transpose();
```
In fact I think could be done like this.

Reply:
It's ok for now